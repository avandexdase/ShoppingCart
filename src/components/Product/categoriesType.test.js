import { render, waitForElement } from '@testing-library/react';
import React from 'react';
import CategoryType from './categoriesType';
import { fireEvent } from '@testing-library/dom';

const data = [
  {
    name: 'Beverages',
    key: 'beverages',
    description:
      'Our beverage department will ensure your fridge is always fully stocked. Shop for soda, juice, beer and more. ',
    enabled: true,
    order: 3,
    imageUrl: '/static/images/category/beverages.png',
    id: '5b675e5e5936635728f9fc30',
  },
  {
    name: 'Bakery Cakes and Dairy',
    key: 'bakery-cakes-dairy',
    description:
      'The best cupcakes, cookies, cakes, pies, cheesecakes, fresh bread, biscotti, muffins, bagels, fresh coffee, milk and more.',
    enabled: true,
    order: 2,
    imageUrl: '/static/images/category/bakery.png',
    id: '5b6899123d1a866534f516de',
  },
];
describe('<categoryType/>', () => {
  const loadCondProductData = jest.fn();
  test('displayed the category', () => {
    const { getByTestId, asFragment } = render(
      <CategoryType
        data={data}
        selectedCategory=""
        loadCondProductData={loadCondProductData}
      />
    );
    const listNode = getByTestId('categoryUl');
    expect(listNode.children).toHaveLength(2);
    expect(asFragment()).toMatchSnapshot();
  });
  test('test onclick of filtered data', () => {
    const { getByTestId, asFragment } = render(
      <CategoryType
        data={data}
        selectedCategory=""
        loadCondProductData={loadCondProductData}
      />
    );
    fireEvent.click(getByTestId('bakery-cakes-dairy'));
    expect(loadCondProductData).toHaveBeenCalledTimes(1)
  });
});
