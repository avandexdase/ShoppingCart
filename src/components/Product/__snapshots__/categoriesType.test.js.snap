// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<categoryType/> displayed the category 1`] = `
<DocumentFragment>
  <div
    class=""
  >
    <div
      class=""
    >
      <div
        class="categoryType__accordian"
        data-testid="categoryAccordian"
      >
        select category
        <div
          class="ExpandMore"
        >
          <svg
            fill="white"
            height="40"
            width="40"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 25.625 10 15.625 11.958 13.667 20 21.708 28.042 13.667 30 15.667Z"
            />
          </svg>
        </div>
      </div>
      <ul
        class="categoryType__ul"
        data-testid="categoryUl"
      >
        <li
          class="categoryType__li"
          data-testid="beverages"
        >
          Beverages
        </li>
        <li
          class="categoryType__li"
          data-testid="bakery-cakes-dairy"
        >
          Bakery Cakes and Dairy
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
